utils/src/main/scala/com/salesforce/op/utils/io/avro/AvroInOut.scala

```diff
@@ -153,27 +155,22 @@ object AvroInOut {
 
   implicit class AvroWriter[T <: GenericRecord](rdd: RDD[T]) {
 
-    private def createJobConfFromContext(schema: String)(implicit sc: SparkSession) = {
-      val jobConf = new JobConf(sc.sparkContext.hadoopConfiguration)
-      AvroJob.setOutputSchema(jobConf, new Schema.Parser().parse(schema))
-      jobConf
-    }
+    private implicit val job = Job.getInstance(rdd.sparkContext.hadoopConfiguration)
 
     /**
      * This method writes out RDDs of generic records as avro files to path.
      *
      * @param path Input directory where avro records should be written.
-     * @param jobConf job config
      * @return
      */
-    def writeAvro(path: String)(implicit jobConf: JobConf): Unit = {
+    def writeAvro(path: String): Unit = {
```

Comment:
please make sure these functions are covered in `AvroInOutTest`, if not add tests.

Reply:
AvroInOutTest has one test for writeAvro already present. Will add some more tests.

Reply:
Made this method private, Since the schema needs to be set before calling this method.

Reply:
I guess so.
AvroKeyOutputFormat  has specific check for this 
if (null == writerSchema) {
      throw new IOException(
          "AvroKeyOutputFormat requires an output schema. Use AvroJob.setOutputKeySchema().");
    }